lms/models/application_instance.py

```diff
@@ -49,36 +39,47 @@ class ApplicationInstance(BASE):
         "OAuth2Token", back_populates="application_instance"
     )
 
+    @classmethod
+    def build_from_lms_url(  # pylint:disable=too-many-arguments
```

Comment:
`build_from_lms_url` is a bit misleading because there are several non-optional parameters here. Perhaps this ought to be named `build` or `build_from_config`?